README.md
[![Release](https://github.com/corellium/tailwind-config/actions/workflows/push.yaml/badge.svg)](https://github.com/corellium/tailwind-config/actions/workflows/push.yaml)

# Tailwind Config Corellium

## Installation

Run the command below to install our Tailwind Config:

```sh
yarn add -D @corellium/tailwind-config
```

Make sure that in target project you have [latest version of tailwind](https://github.com/tailwindlabs/tailwindcss/releases) installed.

```sh
yarn add -D tailwindcss
```

## Usage

In your project root create `tailwind.config.js` file and add theme to the config.

```js
const {
  config: { theme, plugins },
} = require('@corellium/tailwind-config');

module.exports = {
  // By this `theme` filed assingment you will have corellium theme enabled by default
  // https://tailwindcss.com/docs/theme#overriding-the-default-theme
  theme,
  content: [
    // https://tailwindcss.com/docs/content-configuration
  ],
  // https://tailwindcss.com/docs/configuration#plugins
  plugins,
  // Rest of your configuration:
  // https://tailwindcss.com/docs/configuration
};
```


CHANGELOG.md
# v1.0.26 (Wed Jun 15 2022)

#### ⚠️ Pushed to `main`

- Fix type issue ([@unknown4unnamed](https://github.com/unknown4unnamed))

#### 🔩 Dependency Updates

- Bump rollup from 2.75.5 to 2.75.6 [#49](https://github.com/corellium/tailwind-config/pull/49) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.24 to 3.1.2 [#50](https://github.com/corellium/tailwind-config/pull/50) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup-plugin-typescript2 from 0.32.0 to 0.32.1 [#51](https://github.com/corellium/tailwind-config/pull/51) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump @tailwindcss/line-clamp from 0.3.1 to 0.4.0 [#37](https://github.com/corellium/tailwind-config/pull/37) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump auto from 10.36.5 to 10.37.1 [#42](https://github.com/corellium/tailwind-config/pull/42) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup from 2.70.1 to 2.75.5 [#46](https://github.com/corellium/tailwind-config/pull/46) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump typescript from 4.6.4 to 4.7.3 [#48](https://github.com/corellium/tailwind-config/pull/48) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump @rollup/plugin-commonjs from 21.0.3 to 21.1.0 [#33](https://github.com/corellium/tailwind-config/pull/33) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup-plugin-typescript2 from 0.31.2 to 0.32.0 [#47](https://github.com/corellium/tailwind-config/pull/47) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.23 to 3.0.24 [#32](https://github.com/corellium/tailwind-config/pull/32) ([@dependabot[bot]](https://github.com/dependabot[bot]))

#### Authors: 2

- [@dependabot[bot]](https://github.com/dependabot[bot])
- Vasyl Semilyak ([@unknown4unnamed](https://github.com/unknown4unnamed))

---

# v1.0.25 (Thu Jun 02 2022)

#### 🐛 Bug Fix

- feat(config): Add typography plugin [#45](https://github.com/corellium/tailwind-config/pull/45) ([@haydenbarnett](https://github.com/haydenbarnett))

#### ⚠️ Pushed to `main`

- Replace corellium/prettier with harmony ([@haydenbarnett](https://github.com/haydenbarnett))

#### 🔩 Dependency Updates

- Bump typescript from 4.6.3 to 4.6.4 [#36](https://github.com/corellium/tailwind-config/pull/36) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump typescript from 4.6.2 to 4.6.3 [#30](https://github.com/corellium/tailwind-config/pull/30) ([@dependabot[bot]](https://github.com/dependabot[bot]))

#### Authors: 2

- [@dependabot[bot]](https://github.com/dependabot[bot])
- Hayden Barnett ([@haydenbarnett](https://github.com/haydenbarnett))

---

# v1.0.24 (Thu Mar 31 2022)

#### ⚠️ Pushed to `main`

- Merge branch 'main' of github.com:corellium/tailwind-config ([@unknown4unnamed](https://github.com/unknown4unnamed))
- README.md updated ([@unknown4unnamed](https://github.com/unknown4unnamed))

#### Authors: 1

- Vasyl Semilyak ([@unknown4unnamed](https://github.com/unknown4unnamed))

---

# v1.0.23 (Thu Mar 31 2022)

#### ⚠️ Pushed to `main`

- Added line-clamp plugin ([@unknown4unnamed](https://github.com/unknown4unnamed))

#### 🔩 Dependency Updates

- Bump auto from 10.34.1 to 10.36.5 [#29](https://github.com/corellium/tailwind-config/pull/29) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump auto from 10.32.6 to 10.34.1 [#26](https://github.com/corellium/tailwind-config/pull/26) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup from 2.69.0 to 2.70.1 [#28](https://github.com/corellium/tailwind-config/pull/28) ([@dependabot[bot]](https://github.com/dependabot[bot]))

#### Authors: 2

- [@dependabot[bot]](https://github.com/dependabot[bot])
- Vasyl Semilyak ([@unknown4unnamed](https://github.com/unknown4unnamed))

---

# v1.0.22 (Mon Mar 14 2022)

#### ⚠️ Pushed to `main`

- Update mono colour palette for accessibility ([@haydenbleasel](https://github.com/haydenbleasel))

#### 🔩 Dependency Updates

- Bump rollup from 2.68.0 to 2.69.0 [#24](https://github.com/corellium/tailwind-config/pull/24) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump typescript from 4.5.5 to 4.6.2 [#25](https://github.com/corellium/tailwind-config/pull/25) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.22 to 3.0.23 [#21](https://github.com/corellium/tailwind-config/pull/21) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup from 2.67.2 to 2.68.0 [#23](https://github.com/corellium/tailwind-config/pull/23) ([@dependabot[bot]](https://github.com/dependabot[bot]))

#### Authors: 2

- [@dependabot[bot]](https://github.com/dependabot[bot])
- Hayden Bleasel ([@haydenbleasel](https://github.com/haydenbleasel))

---

# v1.0.21 (Tue Feb 15 2022)

#### ⚠️ Pushed to `main`

- Assign dependabot PRs to @haydenbarnett ([@haydenbleasel](https://github.com/haydenbleasel))

#### 🔩 Dependency Updates

- Bump rollup from 2.67.1 to 2.67.2 [#19](https://github.com/corellium/tailwind-config/pull/19) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.19 to 3.0.22 [#20](https://github.com/corellium/tailwind-config/pull/20) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup-plugin-typescript2 from 0.31.1 to 0.31.2 [#18](https://github.com/corellium/tailwind-config/pull/18) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump rollup from 2.66.1 to 2.67.1 [#17](https://github.com/corellium/tailwind-config/pull/17) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.18 to 3.0.19 [#16](https://github.com/corellium/tailwind-config/pull/16) ([@dependabot[bot]](https://github.com/dependabot[bot]))
- Bump tailwindcss from 3.0.16 to 3.0.18 [#14](https://github.com/corellium/tailwind-config/pull/14) ([@dependabot[bot]](https://github.com/dependabot[bot]))

index.js
const hexToRGBA = require('hex-to-rgba');

module.exports = {
  theme: {
    screens: {
      sm: '768px',
      md: '1025px',
      lg: '1280px',
      xl: '1440px',
    },
    colors: {
      transparent: 'transparent',
      current: 'currentColor',
      unset: 'unset',
      inherit: 'inherit',
      white: '#fff',
      space: '#080c14',
      black: '#000',
      gray: {
        DEFAULT: '#475467',
        25: '#fcfcfd',
        50: '#f9fafb',
        100: '#f2f4f7',
        200: '#e4e7ec',
        300: '#d0d5dd',
        400: '#98a2b3',
        500: '#667085',
        600: '#475467',
        700: '#344054',
        800: '#1d2939',
        900: '#101828',
      },
      primary: {
        DEFAULT: '#6938ef',
        25: '#fafaff',
        50: '#f4f3ff',
        100: '#ebe9fe',
        200: '#d9d6fe',
        300: '#bdb4fe',
        400: '#9b8afb',
        500: '#7a5af8',
        600: '#6938ef',
        700: '#5925dc',
        800: '#4a1fb8',
        900: '#3e1c96',
      },
      error: {
        DEFAULT: '#d92d20',
        25: '#fffbfa',
        50: '#fef3f2',
        100: '#fee4e2',
        200: '#fecdca',
        300: '#fda29b',
        400: '#f97066',
        500: '#f04438',
        600: '#d92d20',
        700: '#b42318',
        800: '#912018',
        900: '#7a271a',
      },
      warning: {
        DEFAULT: '#dc6803',
        25: '#fffcf5',
        50: '#fffaeb',
        100: '#fef0c7',
        200: '#fedf89',
        300: '#fec84b',
        400: '#fdb022',
        500: '#f79009',
        600: '#dc6803',
        700: '#b54708',
        800: '#93370d',
        900: '#7a2e0e',
      },
      success: {
        DEFAULT: '#039855',
        25: '#f6fef9',
        50: '#ecfdf3',
        100: '#d1fadf',
        200: '#a6f4c5',
        300: '#6ce9a6',
        400: '#32d583',
        500: '#12b76a',
        600: '#039855',
        700: '#027a48',
        800: '#05603a',
        900: '#054f31',
      },
      blueGray: {
        DEFAULT: '#3e4784',
        25: '#fcfcfd',
        50: '#f8f9fc',
        100: '#eaecf5',
        200: '#c8cce5',
        300: '#9ea5d1',
        400: '#717bbc',
        500: '#4e5ba6',
        600: '#3e4784',
        700: '#363f72',
        800: '#293056',
        900: '#101323',
      },
      blueLight: {
        DEFAULT: '#0086c9',
        25: '#f5fbff',
        50: '#f0f9ff',
        100: '#e0f2fe',
        200: '#b9e6fe',
        300: '#7cd4fd',
        400: '#36bffa',
        500: '#0ba5ec',
        600: '#0086c9',
        700: '#026aa2',
        800: '#065986',
        900: '#0b4a6f',
      },
      blue: {
        DEFAULT: '#1570ef',
        25: '#f5faff',
        50: '#eff8ff',
        100: '#d1e9ff',
        200: '#b2ddff',
        300: '#84caff',
        400: '#53b1fd',
        500: '#2e90fa',
        600: '#1570ef',
        700: '#175cd3',
        800: '#1849a9',
        900: '#194185',
      },
      indigo: {
        DEFAULT: '#444ce7',
        25: '#f5f8ff',
        50: '#eef4ff',
        100: '#e0eaff',
        200: '#c7d7fe',
        300: '#a4bcfd',
        400: '#8098f9',
        500: '#6172f3',
        600: '#444ce7',
        700: '#3538cd',
        800: '#2d31a6',
        900: '#2d3282',
      },
      pink: {
        DEFAULT: '#dd2590',
        25: '#fef6fb',
        50: '#fdf2fa',
        100: '#fce7f6',
        200: '#fcceee',
        300: '#faa7e0',
        400: '#f670c7',
        500: '#ee46bc',
        600: '#dd2590',
        700: '#c11574',
        800: '#9e165f',
        900: '#851651',
      },
      rose: {
        DEFAULT: '#e31b54',
        25: '#fff5f6',
        50: '#fff1f3',
        100: '#ffe4e8',
        200: '#fecdd6',
        300: '#fea3b4',
        400: '#fd6f8e',
        500: '#f63d68',
        600: '#e31b54',
        700: '#c01048',
        800: '#a11043',
        900: '#89123e',
      },
      orange: {
        DEFAULT: '#ec4a0a',
        25: '#fffaf5',
        50: '#fff6ed',
        100: '#ffead5',
        200: '#fddcab',
        300: '#feb273',
        400: '#fd853a',
        500: '#fb6514',
        600: '#ec4a0a',
        700: '#c4320a',
        800: '#9c2a10',
        900: '#7e2410',
      },
    },
    fontFamily: {
      display: [
        'Inter',
        'system-ui',
        '-apple-system',
        'Segoe UI',
        'Roboto',
        'Helvetica Neue',
        'Arial',
        'Noto Sans',
        'Liberation Sans',
        'sans-serif',
        'Apple Color Emoji',
        'Segoe UI Emoji',
        'Segoe UI Symbol',
        'Noto Color Emoji',
      ],
      text: [
        'Inter',
        'system-ui',
        '-apple-system',
        'Segoe UI',
        'Roboto',
        'Helvetica Neue',
        'Arial',
        'Noto Sans',
        'Liberation Sans',
        'sans-serif',
        'Apple Color Emoji',
        'Segoe UI Emoji',
        'Segoe UI Symbol',
        'Noto Color Emoji',
      ],
      mono: [
        'IBM Plex Mono',
        'SFMono-Regular',
        'Menlo',
        'Monaco',
        'Consolas',
        'Liberation Mono',
        'Courier New',
        'monospace',
      ],
    },
    spacing: {
      px: '1px',
      0: '0px',
      1: '0.25rem',
      1.5: '0.375rem',
      2: '0.5rem',
      3: '0.75rem',
      4: '1rem',
      5: '1.25rem',
      6: '1.5rem',
      7: '1.75rem',
      8: '2rem',
      9: '2.25rem',
      10: '2.5rem',
      11: '2.75rem',
      12: '3rem',
      14: '3.5rem',
      16: '4rem',
      20: '5rem',
      24: '6rem',
      28: '7rem',
      32: '8rem',
      36: '9rem',
      40: '10rem',
      44: '11rem',
      48: '12rem',
      52: '13rem',
      56: '14rem',
      60: '15rem',
      64: '16rem',
    },
    fontSize: {
      'display-xs': ['1.25rem', '1.33'],
      'display-sm': ['1.5rem', '1.26'],
      'display-md': ['1.75rem', '1.22'],
      'display-lg': ['2rem', '1.25'],
      'display-xl': ['2.5rem', '1.2'],
      'display-xxl': ['3rem', '1.25'],
      'display-xs-desktop': ['1.5rem', '1.33'],
      'display-sm-desktop': ['1.875rem', '1.26'],
      'display-md-desktop': ['2.25rem', '1.22'],
      'display-lg-desktop': ['3rem', '1.25'],
      'display-xl-desktop': ['3.75rem', '1.2'],
      'display-xxl-desktop': ['4.5rem', '1.25'],
      xs: ['0.75rem', '1.5'],
      sm: ['0.875rem', '1.5'],
      md: ['1rem', '1.5'],
      lg: ['1.125rem', '1.5'],
      xl: ['1.25rem', '1.5'],
    },
    fontWeight: {
      normal: 400,
      medium: 500,
      semibold: 600,
      bold: 700,
    },
    backdropBlur: {
      sm: '0.5rem',
      md: '1rem',
      lg: '1.5rem',
      xl: '2rem',
      xxl: '2.5rem',
    },
    boxShadow: (theme) => ({
      border: `inset 0 0 0 1px ${theme('colors.gray.300')}`,
      'border-light': `inset 0 0 0 1px ${theme('colors.gray.200')}`,
      'border-focus': `inset 0 0 0 1px ${theme('colors.primary.300')}`,
      'border-error': `inset 0 0 0 1px ${theme('colors.error.300')}`,
      focus: `0 0 0 0.25rem ${theme('colors.primary.100')}`,
      'focus-error': `0 0 0 0.25rem ${theme('colors.error.100')}`,
      xs: `0 1px 0.125rem ${hexToRGBA(theme('colors.gray.900'), 0.05)}`,
      sm: `0 1px 0.1875rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 1px 0.125rem ${hexToRGBA(theme('colors.gray.900'), 0.06)}`,
      md: `0 0.25rem 0.5rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.125rem 0.25rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.06
      )}`,
      lg: `0 0.75rem 1rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.25rem 0.375rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.05
      )}`,
      xl: `0 1.25rem 1.5rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.5rem 0.5rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.04
      )}`,
      '2xl': `0 1.75rem 2rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 1rem 0.625rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.03
      )}`,
      '3xl': `0 2.25rem 2.5rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 2rem 0.75rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.02
      )}`,
    }),
    extend: {
      animation: {
        fadeIn: 'fadeIn 0.2s ease-out',
        fadeOut: 'fadeOut 0.2s ease-in',
        enter: 'enter 0.2s ease-out',
        enterCenter: 'enterCenter 0.2s ease-out',
        exit: 'exit 0.2s ease-in forwards',
        exitCenter: 'exitCenter 0.2s ease-in forwards',
      },
      keyframes: {
        fadeIn: {
          '0%': { opacity: 0 },
          '100%': { opacity: 1 },
        },
        fadeOut: {
          '0%': { opacity: 1 },
          '100%': { opacity: 0 },
        },
        enter: {
          '0%': { opacity: 0, transform: 'scale(.9)' },
          '100%': { opacity: 1, transform: 'scale(1)' },
        },
        enterCenter: {
          '0%': { opacity: 0, transform: 'translate(-50%, -50%) scale(.9)' },
          '100%': { opacity: 1, transform: 'translate(-50%, -50%) scale(1)' },
        },
        exit: {
          '0%': { opacity: 1, transform: 'scale(1)' },
          '100%': { opacity: 0, transform: 'scale(.9)' },
        },
        exitCenter: {
          '0%': { opacity: 1, transform: 'translate(-50%, -50%) scale(1)' },
          '100%': { opacity: 0, transform: 'translate(-50%, -50%) scale(0.9)' },
        },
      },
    },
  },
};


rollup.config.js
import typescript from 'rollup-plugin-typescript2';
import commonjs from '@rollup/plugin-commonjs';

const pkg = require('./package.json');

const externalPackages = [...Object.keys(pkg.dependencies || {})];

const regexesOfPackages = externalPackages.map((packageName) => {
  return new RegExp(`^${packageName}(\/.*)?`);
});

const plugins = [
  typescript({
    useTsconfigDeclarationDir: false,
    tsconfig: './tsconfig.json',
  }),
  commonjs(),
];

/**
 * @type {import('rollup').RollupOptions}
 */
const esConfig = {
  input: 'index.ts',
  output: {
    file: 'build/index.js',
    format: 'module',
  },
  plugins,
  external: regexesOfPackages,
};

/**
 * @type {import('rollup').RollupOptions}
 */
const cjsConfig = {
  input: 'index.ts',
  output: {
    file: 'build/index.cjs',
    exports: 'named',
    format: 'commonjs',
    esModule: true,
  },
  plugins,
  external: regexesOfPackages,
};

export default [esConfig, cjsConfig];


index.ts
import hexToRGBA from 'hex-to-rgba';
// NOTE: Need to add .js to fix the issue
import plugin from 'tailwindcss/plugin.js';
import lineClampPlugin from '@tailwindcss/line-clamp';
import typographyPlugin from '@tailwindcss/typography';

export const config = {
  theme: {
    screens: {
      sm: '768px',
      md: '1025px',
      lg: '1280px',
      xl: '1440px',
    },
    colors: {
      transparent: 'transparent',
      current: 'currentColor',
      unset: 'unset',
      inherit: 'inherit',
      white: '#fff',
      black: '#000',
      gray: {
        DEFAULT: '#5e6673',
        25: '#fcfcfd',
        50: '#f9fafb',
        100: '#f2f4f7',
        200: '#e4e7ec',
        300: '#d0d5dd',
        400: '#99a1af',
        500: '#6c7384',
        600: '#4d5665',
        700: '#363c49',
        800: '#252a34',
        900: '#080c14',
      },
      primary: {
        DEFAULT: '#6938ef',
        25: '#fafaff',
        50: '#f4f3ff',
        100: '#ebe9fe',
        200: '#d9d6fe',
        300: '#bdb4fe',
        400: '#9b8afb',
        500: '#7a5af8',
        600: '#6938ef',
        700: '#5925dc',
        800: '#4a1fb8',
        900: '#3e1c96',
      },
      error: {
        DEFAULT: '#d92d20',
        25: '#fffbfa',
        50: '#fef3f2',
        100: '#fee4e2',
        200: '#fecdca',
        300: '#fda29b',
        400: '#f97066',
        500: '#f04438',
        600: '#d92d20',
        700: '#b42318',
        800: '#912018',
        900: '#7a271a',
      },
      warning: {
        DEFAULT: '#dc6803',
        25: '#fffcf5',
        50: '#fffaeb',
        100: '#fef0c7',
        200: '#fedf89',
        300: '#fec84b',
        400: '#fdb022',
        500: '#f79009',
        600: '#dc6803',
        700: '#b54708',
        800: '#93370d',
        900: '#7a2e0e',
      },
      success: {
        DEFAULT: '#039855',
        25: '#f6fef9',
        50: '#ecfdf3',
        100: '#d1fadf',
        200: '#a6f4c5',
        300: '#6ce9a6',
        400: '#32d583',
        500: '#12b76a',
        600: '#039855',
        700: '#027a48',
        800: '#05603a',
        900: '#054f31',
      },
      blueGray: {
        DEFAULT: '#3e4784',
        25: '#fcfcfd',
        50: '#f8f9fc',
        100: '#eaecf5',
        200: '#c8cce5',
        300: '#9ea5d1',
        400: '#717bbc',
        500: '#4e5ba6',
        600: '#3e4784',
        700: '#363f72',
        800: '#293056',
        900: '#101323',
      },
      blueLight: {
        DEFAULT: '#0086c9',
        25: '#f5fbff',
        50: '#f0f9ff',
        100: '#e0f2fe',
        200: '#b9e6fe',
        300: '#7cd4fd',
        400: '#36bffa',
        500: '#0ba5ec',
        600: '#0086c9',
        700: '#026aa2',
        800: '#065986',
        900: '#0b4a6f',
      },
      blue: {
        DEFAULT: '#1570ef',
        25: '#f5faff',
        50: '#eff8ff',
        100: '#d1e9ff',
        200: '#b2ddff',
        300: '#84caff',
        400: '#53b1fd',
        500: '#2e90fa',
        600: '#1570ef',
        700: '#175cd3',
        800: '#1849a9',
        900: '#194185',
      },
      indigo: {
        DEFAULT: '#444ce7',
        25: '#f5f8ff',
        50: '#eef4ff',
        100: '#e0eaff',
        200: '#c7d7fe',
        300: '#a4bcfd',
        400: '#8098f9',
        500: '#6172f3',
        600: '#444ce7',
        700: '#3538cd',
        800: '#2d31a6',
        900: '#2d3282',
      },
      pink: {
        DEFAULT: '#dd2590',
        25: '#fef6fb',
        50: '#fdf2fa',
        100: '#fce7f6',
        200: '#fcceee',
        300: '#faa7e0',
        400: '#f670c7',
        500: '#ee46bc',
        600: '#dd2590',
        700: '#c11574',
        800: '#9e165f',
        900: '#851651',
      },
      rose: {
        DEFAULT: '#e31b54',
        25: '#fff5f6',
        50: '#fff1f3',
        100: '#ffe4e8',
        200: '#fecdd6',
        300: '#fea3b4',
        400: '#fd6f8e',
        500: '#f63d68',
        600: '#e31b54',
        700: '#c01048',
        800: '#a11043',
        900: '#89123e',
      },
      orange: {
        DEFAULT: '#ec4a0a',
        25: '#fffaf5',
        50: '#fff6ed',
        100: '#ffead5',
        200: '#fddcab',
        300: '#feb273',
        400: '#fd853a',
        500: '#fb6514',
        600: '#ec4a0a',
        700: '#c4320a',
        800: '#9c2a10',
        900: '#7e2410',
      },
    },
    fontFamily: {
      display: [
        'Inter',
        'system-ui',
        '-apple-system',
        'Segoe UI',
        'Roboto',
        'Helvetica Neue',
        'Arial',
        'Noto Sans',
        'Liberation Sans',
        'sans-serif',
        'Apple Color Emoji',
        'Segoe UI Emoji',
        'Segoe UI Symbol',
        'Noto Color Emoji',
      ],
      text: [
        'Inter',
        'system-ui',
        '-apple-system',
        'Segoe UI',
        'Roboto',
        'Helvetica Neue',
        'Arial',
        'Noto Sans',
        'Liberation Sans',
        'sans-serif',
        'Apple Color Emoji',
        'Segoe UI Emoji',
        'Segoe UI Symbol',
        'Noto Color Emoji',
      ],
      mono: [
        'IBM Plex Mono',
        'SFMono-Regular',
        'Menlo',
        'Monaco',
        'Consolas',
        'Liberation Mono',
        'Courier New',
        'monospace',
      ],
    },
    spacing: {
      px: '1px',
      0: '0px',
      1: '0.25rem',
      1.5: '0.375rem',
      2: '0.5rem',
      3: '0.75rem',
      4: '1rem',
      5: '1.25rem',
      6: '1.5rem',
      7: '1.75rem',
      8: '2rem',
      9: '2.25rem',
      10: '2.5rem',
      11: '2.75rem',
      12: '3rem',
      14: '3.5rem',
      16: '4rem',
      20: '5rem',
      24: '6rem',
      28: '7rem',
      32: '8rem',
      36: '9rem',
      40: '10rem',
      44: '11rem',
      48: '12rem',
      52: '13rem',
      56: '14rem',
      60: '15rem',
      64: '16rem',
    },
    fontSize: {
      'display-xs': ['1.25rem', '1.33'],
      'display-sm': ['1.5rem', '1.26'],
      'display-md': ['1.75rem', '1.22'],
      'display-lg': ['2rem', '1.25'],
      'display-xl': ['2.5rem', '1.2'],
      'display-xxl': ['3rem', '1.25'],
      'display-xs-desktop': ['1.5rem', '1.33'],
      'display-sm-desktop': ['1.875rem', '1.26'],
      'display-md-desktop': ['2.25rem', '1.22'],
      'display-lg-desktop': ['3rem', '1.25'],
      'display-xl-desktop': ['3.75rem', '1.2'],
      'display-xxl-desktop': ['4.5rem', '1.25'],
      xs: ['0.75rem', '1.5'],
      sm: ['0.875rem', '1.42857142857'],
      md: ['1rem', '1.5'],
      lg: ['1.125rem', '1.55555555556'],
      xl: ['1.25rem', '1.5'],
    },
    fontWeight: {
      normal: 400,
      medium: 500,
      semibold: 600,
      bold: 700,
    },
    backdropBlur: {
      sm: '0.5rem',
      md: '1rem',
      lg: '1.5rem',
      xl: '2rem',
      xxl: '2.5rem',
    },
    boxShadow: (theme: (val: string) => string) => ({
      '4-primary-600': `0 0 0 0.25rem ${theme('colors.primary.600')}`,
      border: `inset 0 0 0 1px ${theme('colors.gray.300')}`,
      'border-light': `inset 0 0 0 1px ${theme('colors.gray.200')}`,
      'border-focus': `inset 0 0 0 1px ${theme('colors.primary.300')}`,
      'border-error': `inset 0 0 0 1px ${theme('colors.error.300')}`,
      focus: `0 0 0 0.25rem ${theme('colors.primary.100')}`,
      'focus-error': `0 0 0 0.25rem ${theme('colors.error.100')}`,
      xs: `0 1px 0.125rem ${hexToRGBA(theme('colors.gray.900'), 0.05)}`,
      sm: `0 1px 0.1875rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 1px 0.125rem ${hexToRGBA(theme('colors.gray.900'), 0.06)}`,
      md: `0 0.25rem 0.5rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.125rem 0.25rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.06
      )}`,
      lg: `0 0.75rem 1rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.25rem 0.375rem -0.125rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.05
      )}`,
      xl: `0 1.25rem 1.5rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 0.5rem 0.5rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.04
      )}`,
      '2xl': `0 1.75rem 2rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 1rem 0.625rem -0.25rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.03
      )}`,
      '3xl': `0 2.25rem 2.5rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.1
      )}, 0 2rem 0.75rem -0.375rem ${hexToRGBA(
        theme('colors.gray.900'),
        0.02
      )}`,
    }),
    extend: {
      animation: {
        fadeIn: 'fadeIn 0.2s ease-out',
        fadeOut: 'fadeOut 0.2s ease-in',
        enter: 'enter 0.2s ease-out',
        enterCenter: 'enterCenter 0.2s ease-out',
        exit: 'exit 0.2s ease-in forwards',
        exitCenter: 'exitCenter 0.2s ease-in forwards',
      },
      keyframes: {
        fadeIn: {
          '0%': { opacity: 0 },
          '100%': { opacity: 1 },
        },
        fadeOut: {
          '0%': { opacity: 1 },
          '100%': { opacity: 0 },
        },
        enter: {
          '0%': { opacity: 0, transform: 'scale(.9)' },
          '100%': { opacity: 1, transform: 'scale(1)' },
        },
        enterCenter: {
          '0%': { opacity: 0, transform: 'translate(-50%, -50%) scale(.9)' },
          '100%': { opacity: 1, transform: 'translate(-50%, -50%) scale(1)' },
        },
        exit: {
          '0%': { opacity: 1, transform: 'scale(1)' },
          '100%': { opacity: 0, transform: 'scale(.9)' },
        },
        exitCenter: {
          '0%': { opacity: 1, transform: 'translate(-50%, -50%) scale(1)' },
          '100%': { opacity: 0, transform: 'translate(-50%, -50%) scale(0.9)' },
        },
      },
    },
  },
  plugins: [
    plugin(({ addBase, addUtilities, theme }) => {
      addBase({
        '*, *::before, *::after': {
          'box-sizing': 'inherit',
        },
        html: {
          'box-sizing': 'border-box',
          'line-height': '1.15',
          'text-size-adjust': '100%',
        },
        body: {
          margin: '0',
          color: theme('colors.gray.900'),
          'font-family': theme('fontFamily.text'),
          '-webkit-font-smoothing': 'antialiased',
          'text-rendering': 'optimizeLegibility',
        },
        'body.dark-magic': {
          color: theme('colors.white'),
        },
        'h1,h2,h3,h4,h5,h6': {
          'font-family': theme('fontFamily.display'),
          'letter-spacing': '-0.02em',
        },
      });
      addUtilities({
        '.gradient-cool': {
          'background-color': '#9b8afb',
          'background-image': `radial-gradient(at 60% 17%, hsl(196deg 99% 71% / 100%) 0, hsl(196deg 99% 71% / 0%) 44%),
            radial-gradient(at 34% 94%, hsl(299deg 96% 75% / 100%) 0, hsl(299deg 96% 75% / 0%) 56%),
            radial-gradient(at 51% 41%, hsl(189deg 66% 78% / 100%) 0, hsl(189deg 66% 78% / 0%) 49%),
            radial-gradient(at 1% 86%, hsl(262deg 62% 63% / 100%) 0, hsl(262deg 62% 63% / 0%) 48%),
            radial-gradient(at 75% 99%, hsl(241deg 86% 66% / 100%) 0, hsl(241deg 86% 66% / 0%) 54%),
            radial-gradient(at 23% 76%, hsl(248deg 95% 64% / 100%) 0, hsl(248deg 95% 64% / 0%) 41%),
            radial-gradient(at 13% 56%, hsl(59deg 61% 64% / 100%) 0, hsl(59deg 61% 64% / 0%) 42%)`,
        },
        '.gradient-warm': {
          'background-color': '#f1a084',
          'background-image': `radial-gradient(at 40% 20%, hsl(28deg 100% 74% / 100%) 0, hsl(28deg 100% 74% / 0%) 50%),
            radial-gradient(at 80% 0%, hsl(189deg 100% 56% / 100%) 0, hsl(189deg 100% 56% / 0%) 50%),
            radial-gradient(at 0% 50%, hsl(355deg 85% 93% / 100%) 0, hsl(355deg 85% 93% / 0%) 50%),
            radial-gradient(at 80% 50%, hsl(340deg 100% 76% / 100%) 0, hsl(340deg 100% 76% / 0%) 50%),
            radial-gradient(at 0% 100%, hsl(22deg 100% 77% / 100%) 0, hsl(22deg 100% 77% / 0%) 50%),
            radial-gradient(at 0% 0%, hsl(343deg 100% 76% / 100%) 0, hsl(343deg 100% 76% / 0%) 50%)`,
        },
      });
    }),
    lineClampPlugin,
    typographyPlugin,
  ],
};

export default { config };
